crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2267,17 +2387,19 @@ impl<'db> Type<'db> {
 
             // Dynamic types are callable, and the return type is the same dynamic type. Similarly,
             // `Never` is always callable and returns `Never`.
-            Type::Dynamic(_) | Type::Never => {
-                Ok(CallOutcome::Single(CallBinding::from_return_type(self)))
-            }
+            Type::Dynamic(_) | Type::Never => OverloadBinding::from_return_type(self)
```

Comment:
This is the one remaining call to `from_return_type`.  Removing it introduces a salsa cycle.

Reply:
Never mind! Picking up the latest `main` lets me remove this too...